### ISSUE TITLE
[Bug] import playlists from Spotify 

### ISSUE THREAD

**Issue by MrSalazar001** [DJDoubleD/refreezer#56]
When I try to import my playlists from Spotify, it returns a error. I placed the Client ID and  the Client Secret correctly but the http://localhost:42069 it's not working.

**Comment by fessoide** [comment]
Yea me too did u resolve?

**Comment by traashful** [comment]
Same here

**Comment by DJDoubleD** [comment]
Does this happen with public playlists as well?
If so, please include an example (not to big) public playlist I can test with.

**Comment by avanteRetro** [comment]
[Changes were made to Spotify developer documentation.](https://developer.spotify.com/documentation/web-api/concepts/redirect_uri)

**Comment by SixMax06** [comment]
Tried with http://127.0.0.1:42069 but it says "INVALID_CLIENT: Invalid redirect URI"

**Comment by tron359** [comment]
Issue occurs due to a new Spotify URL security requirement.

Spotify now blocks all http 'localhost' addresses from use, prefers explicit loopback 127.0.0.1 or https now.
See their dev post at: https://developer.spotify.com/documentation/web-api/concepts/redirect_uri

**Comment by tron359** [comment]
Confirming using http://127.0.0.1:42069/ returns the same message as SixMax06 above, and tracks fail to import.

**Comment by JustASpeedrunner** [comment]
For some reason, the link you're sent to when you click Authorize still uses localhost even if you set the callback to 127.0.0.1. You can edit the link and change where it says localhost to the ip, but even after confirming through the Spotify UI and getting the message, "You can close this page and go back to ReFreezer," and returning to ReFreezer, it loads infinitely.

**Comment by JustASpeedrunner** [comment]
After taking a very brief look into the code, I think it may be caused by [this line here.](https://github.com/DJDoubleD/refreezer/blob/2bb8009bda5898fec3cb59aebb93376288bcc5c1/lib/api/spotify.dart#L178)

I'm willing to bet if this line was changed to the following, it would work as intended.
`const redirectUri = 'http://127.0.0.1:42069';`

**Comment by DJDoubleD** [comment]
> After taking a very brief look into the code, I think it may be caused by [this line here.](https://github.com/DJDoubleD/refreezer/blob/2bb8009bda5898fec3cb59aebb93376288bcc5c1/lib/api/spotify.dart#L178)
> 
> I'm willing to bet if this line was changed to the following, it would work as intended. `const redirectUri = 'http://127.0.0.1:42069';`

Thank you for giving the tip.
I'll try it out as soon as I get a chance.

In the last test I did with the import, it worked with just changing the spotify settings, but my playlist was limited to like 50 tracks that actually got imported (from like 200), so there is at least something extra going on there (just writing it down here for when I get around to debugging this again ;) l

**Comment by ITzERI** [comment]
any updates to this issue?

**Comment by rotmh** [comment]
What's the status of this issue?

**Comment by cerebro84** [comment]
Yes I think it's just a matter or replacing localhost with 127.0.0.1 ; spotify expects https for everything except 127.0.0.1 now, so it's the only workaround

**Comment by JoLander** [comment]
This app works very well for me, especially with Android Auto. It would be even better if I could use my older playlists from Spotify if this issue could be fixed...
